sycl/unittests/thread_safety/ThreadUtils.h

```diff
@@ -0,0 +1,42 @@
+#include <thread>
+#include <vector>
+
+class ParallelTask;
+
+class Thread {
+public:
+  Thread(ParallelTask *Ptr) : MTask(Ptr) {}
+  void start(size_t id);
+  void wait();
+  void body(size_t id);
+
+private:
+  std::thread MThread;
+  ParallelTask *MTask;
+};
+
+class ThreadPool {
+public:
+  ThreadPool(ParallelTask *p);
+  void initialize(int size);
+  void start();
+  void wait();
+
+private:
+  std::vector<Thread *> MThreadPool;
+  ParallelTask *MTask;
+};
```

Comment:
As far as I understand, the user must instantiate only a ParallelTask class and start doing their messy things in multiple threads via `ParallelTask::taskBody` method impl initiated with `ParallelTask::execute`.
Thus, these classes should not be disclosured to the user, should they?

Reply:
Utils were updated.